llvm/lib/CodeGen/ReachingDefAnalysis.cpp

```diff
@@ -48,12 +54,28 @@ static bool isValidRegDefOf(const MachineOperand &MO, MCRegister PhysReg,
   return TRI->regsOverlap(MO.getReg(), PhysReg);
 }
 
+static bool isFIDef(const MachineInstr &MI, int FrameIndex,
+                    const TargetInstrInfo *TII) {
+  int DefFrameIndex = 0;
+  int SrcFrameIndex = 0;
+  if (TII->isStoreToStackSlot(MI, DefFrameIndex) ||
+      TII->isStackSlotCopy(MI, DefFrameIndex, SrcFrameIndex)) {
```

Comment:
Have we tested when isStackSlotCopy is true? I don't know that we implement this on RISC-V backend?

Reply:
> Have we tested when isStackSlotCopy is true? I don't know that we implement this on RISC-V backend?

I don't see it implemeted on RISCV either.

Reply:
Is it needed, if it is not implemented on RISC-V and untested on all other targets?

Reply:
I'll add a test for SystemZ which is the only target which has stack-slot copies

Reply:
Added test for stack-slot copies.